### PR TITLE
roachtest: run pg_regress on insecure clusters

### LINKED ISSUE #119063: roachtest: pg_regress failed
roachtest.pg_regress [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13935478?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13935478?buildTab=artifacts#/pg_regress) on master @ [b2e31876366324c2ebe5c2ad8bbd644997e90864](https://github.com/cockroachdb/cockroach/commits/b2e31876366324c2ebe5c2ad8bbd644997e90864):

```
(pg_regress.go:128).initPGRegress: full command output in run_135202.737205176_n1_psql-postgresroot101.log: COMMAND_PROBLEM: exit status 127
test artifacts and logs in: /artifacts/pg_regress/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13935478/pg-regress/1707573085055/1707573259631)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pg_regress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-36014

### PR DESCRIPTION
**Pull request by DarrylWong**
pg_regress does not support ssl connections so we must turn off secure clusters.

Release note: none
Epic: none
Fixes: #119063

### CODE REVIEW

**Review by rharding6373 — APPROVED** [review]
:lgtm: Thanks for finding this!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/119136#-:-NqYRl8r3QkmguIB1dYV:bxjs5vx)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @herkolategan and @renatolabs)

**Comment by renatolabs** [comment]
I was a bit intrigued by this because the error reported in the original issue was:

```
COMMAND_PROBLEM: exit status 127
```

which, in UNIX-speak, means "command not found".

Indeed, if we look at the output of the command that failed, we see:

```
run_135202.737205176_n1_psql-postgresroot101: 13:52:02 cluster.go:2319: > psql postgres://root@10.142.0.95:29000?sslcert=%2Ftmp%2Froachtest-certs4141163013%2Fcerts%2Fclient.root.crt&sslkey=%2Ftmp%2Froachtest-certs4141163013%2Fcerts%2Fclient.root.key&sslmode=verify-full&sslrootcert=%2Ftmp%2Froachtest-certs4141163013%2Fcerts%2Fca.crt -a -f /mnt/data1/postgres/src/test/regress/sql/test_setup.sql
bash: line 1: -a: command not found
```

`-a: command not found`. I'm pretty sure the reason for this is that we're not quoting the pgurl passed to `psql` and that connection URL has symbols with special meaning (such as `&`).

It also jumps at me that we're using the `/tmp/roachtest-certs4141163013` directory in the connection URL. This won't work because the client asked for an `Internal` URL, and that path only exists in the TeamCity agent (i.e., for `External` PG URLs).

****

TLDR: it's fine to update the test to insecure mode to let the test run, but I think we have some real issues to fix here. I think it should be possible to run this in secure mode once those are fixed.

**Comment by DarrylWong** [comment]
Ah sorry, I should've been more clear on what wasn't working. The failure you're looking at is in the test setup.

>I'm pretty sure the reason for this is that we're not quoting the pgurl passed to psql

Yeah that's correct, I did catch this and try quoting the pgurl. This made it parse it fine, but then it ran into the other issue you mentioned. A side note, but I don't think I touched the `InternalPGUrl` logic for certs at all, so I think it's just always been wrong and this is the first time we've tried it.

I opted to avoid certs all together and use user/password auth. This made the setup happy but the issue comes when we try to run the pg_regress test suite itself.

https://github.com/cockroachdb/cockroach/blob/a30896acefd0da1f92d90b70515dfa7e54578b67/pkg/cmd/roachtest/tests/pg_regress.go#L385-L396

The test suite doesn't have an flag option to pass in any form of certs or password. Without that, all connections get rejected.


So I think you're right that I need to fix `InternalPGUrl`, but I think we still need to run this test on insecure mode.
